gateway/api/repositories/jobs.py

```diff
@@ -0,0 +1,34 @@
+"""
+Repository implementation for Job model
+"""
+import logging
+from django.db.models import Q
+from api.models import Job
+
+logger = logging.getLogger("gateway")
+
+
+class JobsRepository:  # pylint: disable=too-few-public-methods
+    """
+    The main objective of this class is to manage the access to the Job model
+    """
+
+    def get_job_by_id(self, job_id: str) -> Job:
+        """
+        Returns the job for the given id:
+
+        Args:
+            id (str): id of the job
+
+        Returns:
+            Job | None: job with the requested id
+        """
+
+        id_criteria = Q(id=job_id)
+
+        result_queryset = Job.objects.filter(id_criteria).first()
```

Comment:
From what I know `Q` purpose in Django is to build complex queries. How in this case the function `get_job_by_id` leaves pretty clear the what we are doing I think with just:

```
result_queryset = Job.objects.filter(id=job_id).first()
```

Would be more than enough. Let me know @paaragon if you prefer that approach.